### PR TITLE
Make changes to allow for Heroku deployment

### PR DESCRIPTION
**Pull request by Jedeu**
**Changes**

1. Add dockerignore and new Dockerfile for deployment to Heroku
2. Add baseURL for client app so that API calls are made properly
3. Changed pg client in Express app to use connectionString to match deployment
4. Removed unnecessary lock files

### CODE REVIEW

File: api/db/index.js [hunk 1]
```diff
@@ -3,19 +3,18 @@ const { Client } = require('pg')
 const dotenv = require('dotenv')
 dotenv.config()
 
+const isProduction = process.env.NODE_ENV === 'production';
+const connectionString = `postgresql://${process.env.POSTGRES_USER}:${process.env.POSTGRES_PASSWORD}@${process.env.POSTGRES_HOST}:${process.env.POSTGRES_PORT}/${process.env.POSTGRES_DB}`;
```
Comment (mobilemelody) [inline api/db/index.js]: Should we consider just having our local environment use the URL format as well?

File: docker-compose.yml [hunk 2]
```diff
@@ -37,8 +37,6 @@ services:
     image: postgres:13
     ports:
       - 5432:5432
-    env_file:
-      - db.env
```
Comment (mobilemelody) [inline docker-compose.yml]: The removal of this seems to be causing issues with connecting to the database, at least when testing locally

**Review by mobilemelody — CHANGES_REQUESTED** [review]
When I tested these changes locally, the API wasn't connecting to the database. When I added back the two lines with the env file in `docker-compose.yml`, that seemed to fix the issue

File: api/db/index.js (hunk 1 above)
Comment (Jedeu) [reply api/db/index.js]: Yup, they're actually the same thing, but Heroku set this value up under the environment variable `DATABASE_URL`, the connectionString is actually a URL.

File: docker-compose.yml (hunk 2 above)
Comment (Jedeu) [reply docker-compose.yml]: Whoops! Good catch. Didn't mean to remove that

File: api/db/index.js (hunk 1 above)
Comment (mobilemelody) [reply api/db/index.js]: Right, I was wondering if we should just set `DATABASE_URL` in our `db.env` file (as opposed to the individual parameters), and then we wouldn't need to check whether we're in production or not

Comment (Jedeu) [reply api/db/index.js]: Oh that makes sense! Good idea, I've just made the change.

**Review by mobilemelody — APPROVED** [review]
Looks good to me!

**Pull request merged**